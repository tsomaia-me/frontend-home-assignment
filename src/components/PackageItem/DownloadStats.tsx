import React, { useMemo } from 'react'
import 'chart.js/auto'
import { Chart } from 'react-chartjs-2'
import { useEnvironment } from '../../hooks'
import { DateTime } from 'luxon'

export interface DownloadStatsProps {
  downloadStats: Array<number>
}

function DownloadStats({ downloadStats }: DownloadStatsProps) {
  const { numberOfDownloadStatsDays } = useEnvironment()
  const labels = useMemo(
    () => Array(numberOfDownloadStatsDays)
      .fill(null)
      .map((_, index) => DateTime.now().minus({ day: numberOfDownloadStatsDays - index + 1 }).toFormat('ccc')),
    [numberOfDownloadStatsDays]
  )
  const totalCount = useMemo(() => downloadStats.reduce((a, b) => a + b, 0), [downloadStats])

  return (
    <div>
      <h3>{downloadStats.length === 0 ? '--' : totalCount} downloads in the last week</h3>
      <Chart
        type="line"
        data={{
          labels,
          datasets: [
            {
              label: '',
              data: downloadStats,
              borderColor: '#3f51b5',
            }
          ]
        }}
      />
    </div>
  )
}

export default DownloadStats
