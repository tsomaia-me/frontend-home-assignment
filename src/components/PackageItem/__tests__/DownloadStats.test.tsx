import React from 'react'
import renderer from 'react-test-renderer'
import DownloadStats from '../DownloadStats'

test('renders DownloadStats correctly', () => {
  const json = renderer.create(<DownloadStats downloadStats={[]}/>).toJSON()
  expect(json).toMatchSnapshot()
})
