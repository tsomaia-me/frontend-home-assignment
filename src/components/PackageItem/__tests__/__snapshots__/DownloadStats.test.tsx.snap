// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DownloadStats correctly 1`] = `
<div>
  <h3>
    --
     downloads in the last week
  </h3>
  <canvas
    height={150}
    role="img"
    width={300}
  />
</div>
`;
